app/src/main/java/com/alexP/assignment1/ui/addContactFragment/AddContactFragment.kt

```diff
@@ -0,0 +1,149 @@
+package com.alexP.assignment1.ui.addContactFragment
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentFactory
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.lifecycleScope
+import com.alexP.assignment1.databinding.FragmentDialogAddContactBinding
+import com.alexP.assignment1.utils.applyWindowInsets
+import com.alexP.assignment1.utils.getValidationResultMessage
+import com.alexP.assignment1.utils.loadCircularImage
+import com.alexp.contactsprovider.data.Contact
+import com.alexp.textvalidation.data.validator.EmailValidator
+import com.alexp.textvalidation.data.validator.EmptyValidator
+import com.alexp.textvalidation.data.validator.base.BaseValidator
+import com.alexp.textvalidation.data.validator.base.ValidationResult
+import kotlinx.coroutines.launch
+
+
+class AddContactFragment(
+    private val onSaveAction: (Contact) -> Unit
+) : DialogFragment() {
+
+    private lateinit var binding: FragmentDialogAddContactBinding
+    private lateinit var viewModel: AddContactViewModel
+
+    private val galleryLauncher =
+        registerForActivityResult(ActivityResultContracts.GetContent()) { uri ->
+            if (uri == null) return@registerForActivityResult
+            try {
+                viewModel.setGalleryUri(uri)
+                binding.imageViewProfileImage.loadCircularImage(uri.toString())
+            } catch (e: Exception) {
+                e.printStackTrace()
+            }
+        }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentDialogAddContactBinding.inflate(inflater, container, false)
+        binding.root.applyWindowInsets()
+
+        viewModel = ViewModelProvider(this)[AddContactViewModel::class.java]
+
+        observeValues()
+        val toolbar = binding.topBar
+        toolbar.setNavigationOnClickListener {
```

Comment:
Concatenate val toolbar declaration and setNavigationOnClickListener function call.